[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[32mINFO    [0m pytest_operator.plugin:plugin.py:734 Connecting to existing model concierge-lxd:testing on unspecified cloud
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHSession received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHTunneler received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[32mINFO    [0m integration.high_availability.conftest:conftest.py:48 Deploying mysql and scaling to 3 units
[32mINFO    [0m juju.model:model.py:2284 Deploying local:mysql-0
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [allocating] waiting: waiting for machine
  mysql/1 [allocating] waiting: waiting for machine
  mysql/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [allocating] waiting: waiting for machine
  mysql/1 [allocating] waiting: waiting for machine
  mysql/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [allocating] waiting: waiting for machine
  mysql/1 [allocating] waiting: waiting for machine
  mysql/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [allocating] waiting: waiting for machine
  mysql/1 [executing] maintenance: Installing MySQL
  mysql/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [allocating] waiting: waiting for machine
  mysql/1 [executing] maintenance: Installing MySQL
  mysql/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [allocating] waiting: waiting for machine
  mysql/1 [executing] maintenance: Setting up cluster node
  mysql/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [executing] maintenance: Installing MySQL
  mysql/1 [idle] active: Primary
  mysql/2 [executing] maintenance: Installing MySQL
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [executing] maintenance: Setting up cluster node
  mysql/1 [executing] active: Primary
  mysql/2 [executing] maintenance: Setting up cluster node
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [executing] waiting: waiting to join the cluster.
  mysql/1 [idle] active: Primary
  mysql/2 [executing] maintenance: joining the cluster
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [executing] active: 
  mysql/1 [executing] active: Primary
  mysql/2 [executing] active: 
[32mINFO    [0m integration.high_availability.conftest:conftest.py:51 Deploying mysql-test-app
[32mINFO    [0m juju.model:model.py:2284 Deploying ch:amd64/jammy/mysql-test-app-82
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql-test-app/0 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql-test-app/0 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql-test-app/0 [idle] waiting: 
[32mINFO    [0m integration.high_availability.conftest:conftest.py:54 Relating mysql with mysql-test-app
[33mWARNING [0m juju.model:model.py:1666 relate is deprecated and will be removed. Use integrate instead.
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [idle] active: 
  mysql/1 [idle] active: Primary
  mysql/2 [idle] active: 
  mysql-test-app/0 [idle] waiting: 
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [executing] active: 
  mysql/1 [executing] active: Primary
  mysql/2 [executing] active: 
[32mINFO    [0m integration.high_availability.conftest:conftest.py:29 Clearing continuous writes
[32mINFO    [0m integration.high_availability.conftest:conftest.py:32 Starting continuous writes
[32mINFO    [0m integration.helpers:helpers.py:483 Retrieving primary unit
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:341 Ensure continuous writes are incrementing
[32mINFO    [0m integration.helpers:helpers.py:483 Retrieving primary unit
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:366 max_written_value=906 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:366 max_written_value=912 on unit mysql/1
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:366 max_written_value=916 on unit mysql/2
[32mINFO    [0m integration.high_availability.test_self_healing_network_cut:test_self_healing_network_cut.py:54 Unit mysql/1 it's on machine juju-a2a437-1 âœ…
[32mINFO    [0m integration.high_availability.test_self_healing_network_cut:test_self_healing_network_cut.py:70 Cutting network for juju-a2a437-1
[32mINFO    [0m integration.high_availability.test_self_healing_network_cut:test_self_healing_network_cut.py:89 Restoring network for juju-a2a437-1
[32mINFO    [0m integration.high_availability.test_self_healing_network_cut:test_self_healing_network_cut.py:114 Waiting for mysql/1 to enter active
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:341 Ensure continuous writes are incrementing
[32mINFO    [0m integration.helpers:helpers.py:483 Retrieving primary unit
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:366 max_written_value=32707 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:366 max_written_value=32713 on unit mysql/1
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:366 max_written_value=32721 on unit mysql/2
[32mINFO    [0m integration.helpers:helpers.py:483 Retrieving primary unit
[32mINFO    [0m integration.helpers:helpers.py:483 Retrieving primary unit
[32mINFO    [0m integration.high_availability.conftest:conftest.py:37 Clearing continuous writes
[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Model status:

Model    Controller     Cloud/Region         Version  SLA          Timestamp
testing  concierge-lxd  localhost/localhost  3.6.8    unsupported  01:52:47Z

App             Version          Status  Scale  Charm           Channel      Rev  Exposed  Message
mysql           8.0.42-0ubun...  active      3  mysql                          0  no       
mysql-test-app  0.0.2            active      1  mysql-test-app  latest/edge   82  no       

Unit               Workload  Agent  Machine  Public address  Ports           Message
mysql-test-app/0*  active    idle   3        10.57.143.252                   
mysql/0*           active    idle   0        10.57.143.236   3306,33060/tcp  
mysql/1            active    idle   1        10.57.143.47    3306,33060/tcp  
mysql/2            active    idle   2        10.57.143.9                     Primary

Machine  State    Address        Inst id        Base          AZ  Message
0        started  10.57.143.236  juju-a2a437-0  ubuntu@22.04      Running
1        started  10.57.143.47   juju-a2a437-1  ubuntu@22.04      Running
2        started  10.57.143.9    juju-a2a437-2  ubuntu@22.04      Running
3        started  10.57.143.252  juju-a2a437-3  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:868 Juju error logs:

machine-1: 01:38:40 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:38:40 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-1: 01:38:40 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-2: 01:40:09 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:40:09 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-2: 01:40:09 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:40:19 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:40:19 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-0: 01:40:19 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:43:17 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:43:17 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-test-app-0: 01:43:17 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-1: 01:46:10 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: api connection broken unexpectedly
unit-mysql-1: 01:46:31 ERROR unit.mysql/1.juju-log Failed to execute mysql-shell command
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysql_vm_helpers.py", line 795, in _run_mysqlsh_script
    output = subprocess.check_output(  # noqa: S603
  File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['charmed-mysql.mysqlsh', '--passwords-from-stdin', '--uri=serverconfig@juju-a2a437-1.lxd:33062', '--python', '-c', "shell.options.set('useWizards', False)\nprint('###')\ncluster = dba.get_cluster('test_cluster')\nprint(cluster.status({'extended': False}))"]' returned non-zero exit status 1.
unit-mysql-1: 01:46:31 ERROR unit.mysql/1.juju-log Failed to get cluster status for test_cluster
unit-mysql-1: 01:46:40 ERROR unit.mysql/1.juju-log Failed to execute mysql-shell command
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysql_vm_helpers.py", line 795, in _run_mysqlsh_script
    output = subprocess.check_output(  # noqa: S603
  File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['charmed-mysql.mysqlsh', '--passwords-from-stdin', '--uri=serverconfig@juju-a2a437-1.lxd:33062', '--python', '-c', "shell.options.set('useWizards', False)\nprint('###')\ncluster = dba.get_cluster('test_cluster')\nprint(cluster.status({'extended': False}))"]' returned non-zero exit status 1.
unit-mysql-1: 01:46:40 ERROR unit.mysql/1.juju-log Failed to get cluster status for test_cluster
unit-mysql-0: 01:50:13 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-mysql-2: 01:50:13 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-mysql-test-app-0: 01:50:18 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-3: 01:52:33 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-2: 01:52:33 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-0: 01:52:33 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent

[32mINFO    [0m pytest_operator.plugin:plugin.py:947 Forgetting model main...